services/scanners/task_parser.go

```diff
@@ -213,16 +218,34 @@ func (p *ParserTask) parseBlockTransactions(block oasis.Block) (err error) {
 			receiver = raw.Body.Account.String()
 		}
 
+		reclaimAmount := quantity.NewQuantity()
+		//Get reclaim amount
+		if txType.Type() == dmodels.TransactionTypeReclaimEscrow {
+			stakeAccount, err := p.stakingAPI.Account(p.ctx, &stakingAPI.OwnerQuery{
+				Height: block.Header.Height,
+				Owner:  raw.Body.Account,
+			})
+			if err != nil {
+				return err
+			}
+
+			reclaimAmount, err = stakeAccount.Escrow.Debonding.StakeForShares(&raw.Body.Shares)
+			if err != nil {
+				return err
+			}
+		}
+
 		dTxs[i] = dmodels.Transaction{
 			BlockLevel: uint64(block.Header.Height),
 			BlockHash:  hex.EncodeToString(block.Hash),
 			Hash:       tx.Hash().String(),
 			Time:       block.Header.Time,
 			//Same field
 			//Todo probably merge to single field
-			Amount:              raw.Body.Amount.ToBigInt().Uint64(),
+			Amount: raw.Body.Amount.ToBigInt().Uint64(),
+
 			EscrowAmount:        raw.Body.Amount.ToBigInt().Uint64(),
-			EscrowReclaimAmount: raw.Body.Shares.ToBigInt().Uint64(),
+			EscrowReclaimAmount: reclaimAmount.ToBigInt().Uint64(),
```

Comment:
It seems useful to add a separate `EscrowReclaimShares` field which would store the exact number of shares in the `ReclaimEscrow` txn.

Note that `reclaimAmount` is only an approximation at the time when the calculation from shares is made. If the validator is slashed for some reason, the actual reclaimed amount would be proportionally smaller since each share would be worth less tokens (i.e. base units).